IdentityCore/src/cache/crypto/MSIDAssymetricKeyPair+StkJwkExtensions.m

```diff
@@ -0,0 +1,151 @@
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#import "MSIDAssymetricKeyPair+StkJwkExtensions.h"
+
+#import "MSIDJsonObject.h"
+#import "NSData+MSIDAssymetricKeyPairExtensions.h"
+
+@implementation MSIDAssymetricKeyPair (StkJwkExtensions)
+
+- (nullable NSString *)generateStkJwk {
+    NSString * modulus;
+    NSString * exponent;
+    if (![self extractRawKeys:&modulus exponent:&exponent])
+    {
+        return nil;
+    }
+    
+    return [self formatAsJsonWebKey:@"RSA" modulus:modulus exponent:exponent];
+}
+
+// MacOS supports exporting public keys as a DER ASN.1 sequence.
+// A stkjwk can be created by exporting the modulus and exponent based on DER ASN.1 standards.
+// Documentation on the byte structure of DER ASN.1 encoding for the purpose of modulus and exponent parsing for an RSA
+// key is as follows:
+//   [1 byte sequence tag] [3 byte longform sequenceLength]
+//     [1 byte integer indicator] [3 byte longform modulusLength] [modulusLength sized modulusValue]
+//     [1 byte integer indicator] [1 byte shortform exponentLength] [exponentLength sized exponentValue]
+// Notes on expecations of this data format:
+//   sequence tag and length: these are used for internal memory management of a DER ASN.1 data structure
+//                 modulus: For a 2048 bit RSA key, if the modulus is encoded as a positive number, a 256 byte public key will be
+//                          prefixed with an one empty byte. This empty byte prefix is for internal memory management, and can be simply
+//                          stripped off for the purposes of exporting the modulus.
+//                exponent: this is always expected to be "AQAB" for RSA public keys
+//   shortform vs longform: two different types of lengths (1 or two bytes) that describe a values are supported in DER ASN.1 encoding:
+//                          - lengths shorter then 127 are simply encoded as a 1 byte int
+//                          - lengths larger then 126 are 3 bytes long broken into two parts: [1 prefix byte set to 127][a 2 byte int]
+// More details on the DER ASN.1 encoding can be found at:
+// https://docs.microsoft.com/en-us/windows/win32/seccertenroll/about-sequence
+- (bool)extractRawKeys:(NSString **)modulus
+              exponent:(NSString **)exponent {
+    // TODO: Code Reviewers: Is there a recommended code pattern for @available for macOS and iOS?
+    if (@available(macOS 10.12, *)) {
+        CFErrorRef error = nil;
+        NSData *publicKey = (NSData *)CFBridgingRelease(SecKeyCopyExternalRepresentation(_publicKeyRef, &error));
+        if (error) {
+            // TODO: Code Reviewers: What is the best way to propogate error information up the stack to get captured in telemetry? Is it MSID_LOG_WITH_CTX?
```

Comment:
What is the best way to propogate error information up the stack to get captured in telemetry? Is it MSID_LOG_WITH_CTX?

Reply:
MSID_LOG propagates it to the logger. We currently don't have integration of telemetry between ObjC<->MSAL yet.

Reply:
Do you have a backlog task tracking telemetry integration? It would be good to know if we can have that codepath lit up before we start using PRT on MacOS, or how long the gap in telemetry would last